### ISSUE TITLE
Improve option hadndling

### ISSUE THREAD

**Issue by NicoRenaud** [NicoRenaud/votca#2]

**Comment by suvayu** [comment]
IIRC, I have done something like this before with YAML configuration[^1].  However I would *strongly advice* against the nested-ness if data validation is an important requirement.

If you forego nested config, adding validation is relatively straightforward along with other qol improvements like serialisation & persistence.

[^1]: https://github.com/calliope-project/calliope/tree/rules/calliope/typedconfig

**Comment by NicoRenaud** [comment]
I fully agree nested xml files are rather horrible. But I'm afraid we won't change that atm. The current solution already reads the default xml files and create nested namespaces out of them. Users can then modify these namespaces and the code recreates the xml files the with user-defined entries ... no ideal but functional 

Modifying the input structure would require touching the C++ code and all the input validation that is done there. We could easily spend the rest of the budget just figuring out how it is done on the C++ side since the main developer (Jens) left us

**Comment by suvayu** [comment]
If by "modifying the namespace" you mean a user can add new keys, as opposed to changing values, validation would be very difficult.  But otherwise, it should be possible to deal with nested-ness, albeit somewhat complicated.

**Comment by NicoRenaud** [comment]
no ideally users should not be able to add new keys but only change their values. At the moment they *can* add new keys and I want to prevent that. The nested-ness is headache inducing but indeed it should be manageable

**Comment by suvayu** [comment]
Since this is Python, my impression is restricting new keys and data validation are two sides of the same coin.  I'll be in the office later, if you are around we can have a hands-on discussion and evaluate if this is the right path.  That said, even though `pydantic` uses database terminology, it's quite a bit simpler than that :-p.
